lib/langchain/vectorsearch/qdrant.rb

```diff
@@ -123,7 +124,7 @@ def ask(question:)
 
       prompt = generate_prompt(question: question, context: context)
 
-      llm.chat(prompt: prompt)
+      llm.chat(prompt: prompt, **@options)
```

Comment:
Instead of passing these options via the `Qdrant#ask` method I would probably recommend actually moving all of this method's logic into your own application:

```ruby
      search_results = similarity_search(query: question)

      context = search_results.map do |result|
        result.dig("payload").to_s
      end
      context = context.join("\n---\n")

      prompt = generate_prompt(question: question, context: context)

      llm.chat_with_functions(prompt: prompt, functions: functions)
```

Reply:
This works like the Langchain::Conversation.new, in the way options can be passed through the LLM from that interface. It is handy as it seems the only way to pass options to the LLM from the vector search, Not only functions but other options too. Why would VectorSearch be different from Conversation in terms of options customization?

Reply:
Okay -- what's your thought on passing the `functions: []` to the `Langchain::Conversation` instance? E.g.:
```ruby
chat = Langchain::Conversation.new(llm: openai) 
chat.set_context("You are the climate bot")
chat.functions = functions
```

Reply:
Definetely adding it as an accessor should work too. But given the `Langchain::Conversation` initializer has an `options` attribute already there I don't see any reason to not use that. You have the last word on this of course. 
Also, I would suggest that as the Conversation has the `options` that can be passed through the LLM it would make sense that the VectorSearch interface also has `options`. That way the API will be unified and will act as a transparent connector without needing to write/port logic to the apps.

Reply:
Would you please add the "chat.functions = functions" to the Conversation class? I don't think vector search classes should know about functions, doesn't seem like it's related.

Reply:
I can add an accessor to the Conversation class, I expect to work on this tomorrow. Just I'm wondering why this does not fit your design needs.

```ruby
chat = Langchain::Conversation.new(llm: openai, functions: functions, complete_chunk: true ) 
```
It actually works as is without adding much code. 

note that we also need the complete_chunk option ... and eventually other options will be added from the open AI. 


Also this:
> I don't think vector search classes should know about functions, it doesn't seem like it's related.

Do you mean that Vector Search should not have support for GPT functions? or function options should not be passed through the Vector Search as options? 
I ask this as I Indeed have some use cases for this. VectorSearch + Function calls.

Reply:
I don't think function options should be passed through the vector search as options.

Reply:
Examples:

1. conversation func calls https://gist.github.com/michelson/9c6ca9f10c39d81af0cc9ab68ec6adc7
2. vector search func calls: https://gist.github.com/michelson/bea785ddb0b3fa4faf9654cbe2373d73

Conversation essential bits are:

```ruby
chat = Langchain::Conversation.new(llm: openai)
chat.set_context("You are the climate bot")
chat.set_functions(functions)
```

Vector search important part is:

```ruby
client.llm.functions = functions
client.llm.complete_chunk = true
chat = client.ask(question: "create a users_controller")
```


Note the complete chunks in the conversation version will be added automatically on the `set_functions.` On the other hand the vector search version, you will need to add this option explicitly.